addons/l10n_id_efaktur/wizard/reexport_taxnumber.py

```diff
@@ -0,0 +1,68 @@
+# -*- encoding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+import re
+
+
+class ReexportTaxnumber(models.TransientModel):
+    _name = 'reexport.taxnumber'
+    _description = "Report Tax-Number"
+
+    start_number = fields.Char('From', compute='_compute_start_number', store=True, readonly=False)
+    end_number = fields.Char('To', default="0000000000000")
+    company_id = fields.Many2one('res.company', default=lambda self: self.env.company)
+    number_available = fields.Integer(compute='_compute_available')
+    last_available = fields.Many2one('l10n_id_efaktur.efaktur', compute='_compute_available')
+
+    def generate_number(self):
+        """Generate tax number."""
+        start_str = re.sub(r'\D', '', self.start_number)
+        end_str = re.sub(r'\D', '', self.end_number)
+
+        start_int = int(start_str)
+        end_int = int(end_str)
+
+        if not len(start_str) == 13 or not len(end_str) == 13:
+            raise ValidationError(_("There should be 13 digits in each number."))
+
+        if start_str[:5] != end_str[:5]:
+            raise ValidationError(_("First 5 digits should be same in Start Number and End Number."))
+
+        if int(start_str[-8:]) >= int(end_str[-8:]):
+            raise ValidationError(_("Last 8 digits of End Number should be greater than the last 8 digit of Start Number"))
+
+        if (end_int - start_int) > 10000:
+            raise ValidationError(_("The difference between the two numbers must not be greater than 10.000"))
+
+        self.env['l10n_id_efaktur.efaktur'].create([{
```

Comment:
Why???  Maybe we had the discussion before but this is so weird...  We can not just store ranges and have a last number?  (With sequences we removed everything and here we go the other extreme)

Reply:
It's not the same as sequences... Here the numbers have a meaning, and they can be modified over time (when you assign it to an invoice, and when you want to modify an invoice already with a number.